docs/topics/native/native-objc-interop.md

```diff
@@ -32,27 +32,134 @@ Kotlin modules can be used in Swift/Objective-C code if compiled into a framewor
+Another alternative is to mark Kotlin declarations with the` internal` modifier, which restricts their access to their
+compilation module. Choose `@HiddenFromObjC` when only wanting to hide the Kotlin declaration from Objective-C and
+Swift, but still be visible from other Kotlin modules.
 
-  [See an example in the Kotlin-Swift interopedia](https://github.com/kotlin-hands-on/kotlin-swift-interopedia/blob/main/docs/overview/HiddenFromObjC.md).
-* `@ShouldRefineInSwift` helps to replace a Kotlin declaration with a wrapper written in Swift. The annotation marks a
-  function or property as `swift_private` in the generated Objective-C API. Such declarations get the `__` prefix,
-  which makes them invisible from Swift.
+[See an example in the Kotlin-Swift interopedia](https://github.com/kotlin-hands-on/kotlin-swift-interopedia/blob/main/docs/overview/HiddenFromObjC.md).
 
-  You can still use these declarations in your Swift code to create a Swift-friendly API, but they won't be suggested in
-  the Xcode autocomplete.
+### Use refining in Swift
 
-  * For more information on refining Objective-C declarations in Swift, see the [official Apple documentation](https://developer.apple.com/documentation/swift/improving-objective-c-api-declarations-for-swift).
-  * For an example on how to use the `@ShouldRefineInSwift` annotation, see the [Kotlin-Swift interopedia](https://github.com/kotlin-hands-on/kotlin-swift-interopedia/blob/main/docs/overview/ShouldRefineInSwift.md).
+> The `@ShouldRefineInSwift` annotation is [Experimental](components-stability.md#stability-levels-explained) and requires [opt-in](opt-in-requirements.md).
+>
+{type="warning"}
+
+`@ShouldRefineInSwift` helps to replace a Kotlin declaration with a wrapper written in Swift. The annotation marks a
+function or property as `swift_private` in the generated Objective-C API. Such declarations get the `__` prefix,
+which makes them invisible from Swift.
+
+You can still use these declarations in your Swift code to create a Swift-friendly API, but they won't be suggested in
+the Xcode autocomplete.
+
+* For more information on refining Objective-C declarations in Swift, see the [official Apple documentation](https://developer.apple.com/documentation/swift/improving-objective-c-api-declarations-for-swift).
+* For an example on how to use the `@ShouldRefineInSwift` annotation, see the [Kotlin-Swift interopedia](https://github.com/kotlin-hands-on/kotlin-swift-interopedia/blob/main/docs/overview/ShouldRefineInSwift.md).
+
+### Change declaration names
+
+> The `@ObjCName` annotation is [Experimental](components-stability.md#stability-levels-explained) and requires [opt-in](opt-in-requirements.md).
+>
+{type="warning"}
+
+To avoid renaming Kotlin declarations, use the `@ObjCName` annotation. It instructs the Kotlin compiler to use
+a custom Objective-C and Swift name for classes, interfaces, and other Kotlin concepts:
```

Comment:
```suggestion
custom Objective-C and Swift names for classes, interfaces, and other Kotlin entities:
```

Reply:
I think we should keep 'concepts' here, this term is used in the first level of the docs tree

Reply:
Oh, I'm not saying that the term is wrong, but that it's not appropriate here: we're trying to say that `MySwiftArray` can be a custom name for the entity called `MyKotlinArray`. `MyKotlinArray` is an entity of the kind described by the Kotlin concept of a class, but that's not important here, as an entity of any kind can be assigned a custom name.

Maybe it's more clear to rewrite the sentence like this: "It instructs the Kotlin compiler to use the specified custom name for the annotated class, interface, or another Kotlin entity." Right now the plural "classes" can indeed be understood both as referring to the concept of a class and to the set of all implemented classes.